runtime/lua/vim/diagnostic.lua

```diff
@@ -853,6 +850,13 @@ function M.goto_next(opts)
   return diagnostic_move_pos(opts, M.get_next_pos(opts))
 end
 
+local get_fallback_sign = function (severity)
+  local sign = vim.fn.sign_getdefined(sign_highlight_map[severity])
+  if sign and sign[1] and not vim.tbl_isempty(sign[1]) then
+    return sign[1].text
+  end
```

Comment:
This check seems a bit over-defensive. We can trust that `sign_getdefined` will return a list (non-nil) and we also can trust that each table entry in the list will have its values populated.

```suggestion
  if sign[1] then
    return sign[1].text
  end
```